### PR TITLE
[SymbolFileDWARF] Remove a swift-specific hack that was working around a bug

### PR DESCRIPTION
**Pull request by dcci**
And cherry-pick related commits.

### CODE REVIEW

**Comment by dcci** [comment]
@swift-ci test

File: lldb/source/Plugins/SymbolFile/DWARF/SymbolFileDWARF.cpp [hunk 1]
```diff
@@ -3620,8 +3620,7 @@ VariableSP SymbolFileDWARF::ParseVariableDIE(const SymbolContext &sc,
             }
           }
         } else {
-          if (location_is_const_value_data &&
-              !IsSwiftLanguage(sc.comp_unit->GetLanguage()))
```
Comment (dcci) [inline lldb/source/Plugins/SymbolFile/DWARF/SymbolFileDWARF.cpp:3624]: Stuffs like this shouldn't exist *at all*

File: lldb/test/Shell/SymbolFile/DWARF/static_scope.s [hunk 2]
```diff
@@ -0,0 +1,312 @@
+# Test that the DWARF parser assigns the right scope to the
+# variable `b`, which is `local` and not `static`.
+
+# REQUIRES: x86
+# UNSUPPORTED: lldb-repro
```
Comment (vedantk) [inline lldb/test/Shell/SymbolFile/DWARF/static_scope.s:5]: What's the deal here (could you leave a brief/1liner comment about why lldb-repro isn't supported)?

File: lldb/test/Shell/SymbolFile/DWARF/static_scope.s [hunk 3]
```diff
@@ -0,0 +1,312 @@
...
+
+# RUN: llvm-mc -triple=x86_64-apple-macosx10.15.0 -filetype=obj %s > %t.o
+# RUN: lldb-test symbols %t.o | FileCheck %s
+
+# CHECK: Variable{{.*}}, name = "b", type = {{.*}} (int), scope = local
+
```
Comment (vedantk) [inline lldb/test/Shell/SymbolFile/DWARF/static_scope.s:11]: Please leave instructions for how to produce the same assembly from source.

**Review by vedantk — COMMENTED** [review]
Generally looks good!

File: lldb/test/Shell/SymbolFile/DWARF/static_scope.s (hunk 2 above)
Comment (dcci) [reply lldb/test/Shell/SymbolFile/DWARF/static_scope.s:5]: This is a cherry-pick from upstream. I guess you might want to comment there.
The only relevant swift bit is the one I outlined before [the check removed].

File: lldb/source/Plugins/SymbolFile/DWARF/DWARFDebugInfoEntry.cpp [hunk 4]
```diff
@@ -1230,6 +1230,29 @@ DWARFDebugInfoEntry::GetAbbreviationDeclarationPtr(const DWARFUnit *cu) const {
   return nullptr;
 }
 
+bool DWARFDebugInfoEntry::IsGlobalOrStaticVariable() const {
+  if (Tag() != DW_TAG_variable)
+    return false;
+  const DWARFDebugInfoEntry *parent_die = GetParent();
+  while (parent_die != nullptr) {
+    switch (parent_die->Tag()) {
+    case DW_TAG_subprogram:
```
Comment (adrian-prantl) [inline lldb/source/Plugins/SymbolFile/DWARF/DWARFDebugInfoEntry.cpp:1239]: This is counter-intuitive. How is
```
void f() {
static int g_i = 0;
}
```
represented in DWARF?

Comment (dcci) [reply lldb/source/Plugins/SymbolFile/DWARF/DWARFDebugInfoEntry.cpp:1239]: Can you please comment on the upstream version? This is a cherry-pick

Comment (adrian-prantl) [reply lldb/source/Plugins/SymbolFile/DWARF/DWARFDebugInfoEntry.cpp:1239]: Done (https://reviews.llvm.org/D77698)

**Comment by dcci** [comment]
Merging this while we continue the conversation upstream.

**Pull request merged**